internal/pkg/vm/actor/builtin/miner/miner.go

```diff
@@ -470,7 +470,7 @@ func (*Impl) AddAsk(ctx invocationContext, price types.AttoFIL, expiry *big.Int)
 
 	var state State
 	out, err := actor.WithState(ctx, &state, func() (interface{}, error) {
-		if ctx.Message().From != state.Worker {
+		if ctx.OriginalMessage().From != state.Worker {
```

Comment:
You can change this to ctx.Caller and remove the message. Same goes for all the other uses of From in this file (and everywhere else).